deepchecks/utils/ipython.py

```diff
@@ -304,121 +267,511 @@ def __exit__(self, *args, **kwargs):
+        return False
+
+    extension = extensions.get(name)
+
+    return (
+        extension.enabled and extension.validate()
+        if extension is not None
+        else False
+    )
+
+
+def get_jupyter_server_extensions() -> t.Optional[t.Mapping[str, t.Any]]:
+    """Get dictionary of jupyter server extensions.
+
+    Returns
+    -------
+    None :
+        when 'jupyter_server' and 'jupyter_core' are not available,
+        it means that 'Jupyter' is not installed
+    Mapping[str, jupyter_server.extension.manager.ExtensionPackage] :
+        map of extension name -> extension package instance
+    """
+    try:
+        from jupyter_core.paths import jupyter_config_path
+        from jupyter_server.extension.config import ExtensionConfigManager
+        from jupyter_server.extension.manager import ExtensionManager
+    except ImportError:
+        return
+    else:
+        folders = t.cast(t.List[str], jupyter_config_path())
+        config_manager = ExtensionConfigManager(read_config_path=folders)
+        extension_manager = ExtensionManager(config_manager=config_manager)
+        return extension_manager.extensions
+
+
+def get_jupyter_server_url() -> t.Optional[str]:
+    """Get running jupyter server url.
+
+    Returns
+    -------
+    None :
+        when there is no running jupyter server instance,
+        or when there is more than one running jupyter
+        server instance. In the second case, we cannot
+        determine which one is ours.
+    str :
+        jupyter server url string
+    """
+    try:
+        output = subprocess.getoutput('jupyter server list').split('\n')
+    except BaseException:
+        return
+    else:
+        urls = [
+            line
+            for line in output
+            if line.strip().startswith('http') or line.strip().startswith('https')
+        ]
+
+        if len(urls) > 1:
+            warnings.warn(
+                'Found more then one running jupyter server. '
+                'Not able to determine which one is used by current user, '
+                'and as result it can effect what kind of output (interactive, '
+                'non-interactive) will be used to display check/suite results'
+            )
+            return
+        if len(urls) == 0:
+            return
+
+        url = urls[0].split('::')[0].strip()
+        return url.split(' ')[0]
+
+
+def extract_jupyter_server_token(url: str) -> str:
+    """Extract token string from jupyter server url query params string."""
+    query = parse_qs(url)
+    token = (query.get('token') or [])
+    return (token[0] if len(token) > 0 else '')
+
+
+def is_jupyterlab_extension_enabled(name: str) -> bool:
+    """Find out whether provided nbclassic extension is enabled."""
+    server_url = get_jupyter_server_url()
+    extensions = None
+
+    if server_url is not None:
+        extensions = request_jupyterlab_extensions(server_url)
+
+    if extensions is None:
+        extensions = get_jupyterlab_extensions()
```

Comment:
is this a more reliable way? why not just always use this and remove the server request?

Reply:
cli command will always take a look at the default config path but
what if the user started a server with some non-default config path?

Reply:
Please add explanation about that in the code

Reply:
added